core/src/main/scala/org/bitcoins/core/crypto/AesCrypt.scala

```diff
@@ -7,54 +7,297 @@ import javax.crypto.{BadPaddingException, Cipher, SecretKey}
 import scodec.bits.ByteVector
 
 import scala.util.{Failure, Success, Try}
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.Factory
 
-case class AesEncryptedData(
-    cipherText: ByteVector,
-    iv: ByteVector,
-    salt: AesSalt)
+/**
+  * Represents a encrypted cipher text with it's accompanying
+  * initialization vector (IV). Both the cipher text and the IV
+  * is needed to decrypt the cipher text.
+  */
+final case class AesEncryptedData(cipherText: ByteVector, iv: AesIV) {
+
+  /**
+    * We serialize IV and ciphertext by prepending the IV
+    * to the ciphertext, and converting it to base64.
+    * Since the IV is of static length, deserializing is a matter
+    * of taking the first bytes as IV, and the rest as
+    * ciphertext.
+    */
+  lazy val toBase64: String = {
+    val bytes = iv.bytes ++ cipherText
+    bytes.toBase64
+  }
+}
 
-case class AesSalt(
-    value: ByteVector
-)
+object AesEncryptedData {
 
-/**
-  * @throws IllegalArgumentException if passed an empty string
+  /**
+    * We serialize IV and ciphertext by prepending the IV
+    * to the ciphertext, and converting it to base64.
+    * Since the IV is of static length, deserializing is a matter
+    * of taking the first bytes as IV, and the rest as
+    * ciphertext.
+    */
+  def fromBase64(base64: String): Option[AesEncryptedData] = {
+    ByteVector.fromBase64(base64) match {
+      case None                                        => None
+      case Some(bytes) if bytes.length <= AesIV.length => None
```

Comment:
So this is implying that you can have a empty cipher text with an IV? That seems to be the assumption any way

Reply:
Added a comment explaining the line